File: Javascript/Algorithms/387.js
/**
 * @param {string} s
 * @return {number}
 */
let firstUniqChar = s => {
    const chars = Array(26).fill(0);
    let idx = 0;
    for (c of s)
        chars[c.codePointAt() - 97]++;
    for (c of s){
        if (chars[c.codePointAt() - 97] == 1)
            return idx;
        idx++;
    }
    return -1;
};

File: Rust/Algorithms/387.rs
impl Solution {
    pub fn first_uniq_char(s: String) -> i32 {
        let mut chars = vec![0; 26];
        let cpoints = s.as_bytes();

        for p in cpoints {
            chars[(p - 97) as usize] += 1;
        }

        let mut idx = 0;
        for p in cpoints {
            if chars[(p - 97) as usize] == 1 {
                return idx;
            }
            idx += 1
        }
        -1
    }
}
